addons/mail/static/src/discuss/core/common/channel_member_model.js

```diff
@@ -33,7 +33,10 @@ export class ChannelMember extends Record {
     channel_id = fields.One("Thread", { inverse: "channel_member_ids" });
     threadAsSelf = fields.One("Thread", {
         compute() {
-            if (this.store.self?.eq(this.persona)) {
+            if (
+                this.store.self_partner?.eq(this.partner_id) ||
+                this.store.self_guest?.eq(this.guest_id)
+            ) {
```

Comment:
As discussed in another PR, comparison are still much easier when keeping the concept of persona.